### PR TITLE
Optional "to" and "from"

### PR DESCRIPTION
**Pull request by fromonesrc**
Can be invoked with `botzo ship app to stage from master` OR just `botzo ship app stage master` because I'm too lazy and stupid :smile: 

@tecnobrat

### CODE REVIEW

File: scripts/deploy.coffee [hunk 1]
```diff
@@ -2,27 +2,24 @@
...
-JENKINS_JOB = process.env.JENKINS_JOB
-JENKINS_USER = process.env.JENKINS_USER
+JENKINS_URL     = process.env.JENKINS_URL
+JENKINS_JOB     = process.env.JENKINS_JOB
+JENKINS_USER    = process.env.JENKINS_USER
 JENKINS_API_KEY = process.env.JENKINS_API_KEY
 
 module.exports = (robot) ->
-  pattern = new RegExp('(?:deploy|ship) (\\\S+) to (\\\S+)(?: from (\\\S+))?$', 'i')
+  pattern = new RegExp("(?:deploy|ship) (\\\S+)( to)? (\\\S+)(?:( from)? (\\\S+))?$", "i")
```
Comment (tecnobrat) [inline scripts/deploy.coffee]: The single quote vs double quote is actually probably very important here, due to the way the `\\\S+` escaping works.  Have you tested this?

**Comment by tecnobrat** [comment]
The only comment I have against this is if people don't put the order correct, and its like:

`deploy one demo ci`

Where they have actually named a branch demo or something.

**Comment by fromonesrc** [comment]
That's a good point. I think most people will use to/from but the super 1337 can omit it if they know what they're doing. Think that's fair?

**Comment by tecnobrat** [comment]
I'm not sure how likely this is, but if people can't remember deploy x to y from z, they may not remember the order either.  Maybe a happy medium to ensure branches / env confusion doesn't happen is to always require `from`?

So these are all valid:

`botzo deploy one ci`
`botzo deploy one to ci`
`botzo deploy one ci from master`
`botzo deploy one to ci from master`

**Comment by tecnobrat** [comment]
That being said, I'm fine to ship this as-is (after my double vs single quote question in the diff is answered).  Just wanted to raise it as a concern and open the discussion.

File: scripts/deploy.coffee (hunk 1 above)
Comment (fromonesrc) [inline scripts/deploy.coffee]: Tested and working

**Comment by tecnobrat** [comment]
@fromonesrc feel free to override me if you are cool with this as-is, it has my :+1:

**Comment by fromonesrc** [comment]
Lemme :thought_balloon: on this some more. You raise some good points. I still think the people who would omit the "to" and "from" in favor of the short hand would be a small group who understood the syntax but I can see how it could cause some :corn:fusion.

**Pull request closed without merging**